neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -19,24 +20,52 @@ namespace Neo.Network.P2P
     internal class ProtocolHandler : UntypedActor
     {
         public class SetFilter { public BloomFilter Filter; }
+        internal class Timer { }
+
+        private class PendingKnownHashesCollection : KeyedCollection<UInt256, KeyValuePair<UInt256, DateTime>>
+        {
+            protected override UInt256 GetKeyForItem(KeyValuePair<UInt256, DateTime> item)
+            {
+                return item.Key;
+            }
+        }
 
         private readonly NeoSystem system;
+        private readonly PendingKnownHashesCollection pendingKnownHashes;
```

Comment:
`Dictionary<UInt256, DateTime>`?

Reply:
This version can serve the the hashes from different clients.

Reply:
What do you mean?

Reply:
Sorry, the PR is long time ago, I forgot some details, so I checked again, It's for easier clean the outdated items. https://github.com/neo-project/neo/blob/de4080b29237d1f1c81e554c6ad8a07aacc51884/neo/Network/P2P/ProtocolHandler.cs#L346

Reply:
OrderedDictionary doesn't help?

Reply:
I think they are the same.

Reply:
> OrderedDictionary doesn't help?

DateTime is a struct type.